### PR TITLE
[HttpKernel] fix deprecating KernelInterface::getRootDir()

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.2
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Follow up of #28810.
Did I get it right?

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
(ready)

File: src/Symfony/Bundle/FrameworkBundle/Templating/Helper/CodeHelper.php [hunk 1]
```diff
@@ -38,8 +33,7 @@ class CodeHelper extends Helper
     public function __construct($fileLinkFormat, string $projectDir, string $charset)
     {
         $this->fileLinkFormat = $fileLinkFormat ?: ini_get('xdebug.file_link_format') ?: get_cfg_var('xdebug.file_link_format');
-        $this->projectDir = str_replace('\\', '/', $projectDir).'/';
-        $this->rootDir = $this->projectDir;
+        $this->rootDir = str_replace('\\', '/', $projectDir).'/';
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Templating/Helper/CodeHelper.php]: I would keep the new name, it's easier to follow along.

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Templating/Helper/CodeHelper.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Bundle/FrameworkBundle/Templating/Helper/CodeHelper.php]: but changin it breaks BC...

Comment (fabpot) [reply src/Symfony/Bundle/FrameworkBundle/Templating/Helper/CodeHelper.php]: Yes and no. That's internal/obsolete code (templating vs Twig).

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**